### PR TITLE
Port DeepBSDE and DeepBSDEHan from NerualPDE

### PR DESCRIPTION
**Pull request by ashutosh-b-b**
- Ports DeepBSDE and DeepBSDEHan from NeuralPDE
- Updates tests and fix bugs. Make compatible with julia 1.9
- Fix existing tests

### CODE REVIEW

**Comment by ChrisRackauckas** [comment]
Do we still need TerminalPDEProblem when there's PIDEProblem? You're just giving a terminal condition and don't support the integral term, right?

**Comment by ashutosh-b-b** [comment]
> Do we still need TerminalPDEProblem when there's PIDEProblem? You're just giving a terminal condition and don't support the integral term, right?

Yes, we can just use the PIDEProblem. fixed that in : https://github.com/SciML/HighDimPDE.jl/pull/68/commits/211c12020b76bfcfdf0d8811fd1f6b7ea368216d

**Comment by ChrisRackauckas** [comment]
Resolve conflict and I think this is good.

File: Project.toml [hunk 1]
```diff
@@ -28,4 +33,4 @@ Functors = "0.2, 0.3, 0.4"
 Reexport = "1"
 Statistics = "1"
 Zygote = "0.6"
-julia = "1.6,1.7"
+julia = "1.8, 1.9"
```
Comment (ChrisRackauckas) [inline Project.toml:36]: this should just be a lower bound. It can be bumped to v1.10

**Comment by ChrisRackauckas** [comment]
@ArnoStrouwen can you do your pass on this library? The CI, version bounds, etc. need a full assessment here.

**Pull request merged**